openapidocs/v31.py

```diff
@@ -0,0 +1,391 @@
+"""
+This module defines classes that can be used to generate OpenAPI Documentation
+version 3.1.
+https://swagger.io/specification/
+"""
+from abc import ABC
+from dataclasses import dataclass
+from enum import Enum
+from typing import Any, Dict, List, Optional, Union
+
+from openapidocs.common import OpenAPIRoot, normalize_dict
+
+from .common import OpenAPIElement
+
+
+class ParameterLocation(Enum):
+    QUERY = "query"
+    HEADER = "header"
+    PATH = "path"
+    COOKIE = "cookie"
+
+
+class ValueType(Enum):
+    ARRAY = "array"
+    BOOLEAN = "boolean"
+    INTEGER = "integer"
+    NUMBER = "number"
+    OBJECT = "object"
+    STRING = "string"
+
+
+class ValueFormat(Enum):
+    BASE64 = "base64"
+    BINARY = "binary"
+    BYTE = "byte"
+    DATE = "date"
+    DATETIME = "date-time"
+    TIME = "time"
+    DURATION = "duration"
+    DOUBLE = "double"
+    FLOAT = "float"
+    INT32 = "int32"
+    INT64 = "int64"
+    PASSWORD = "password"
+    EMAIL = "email"
+    IDNEMAIL = "idn-email"
+    UUID = "uuid"
+    PARTIALTIME = "partial-time"
+    HOSTNAME = "hostname"
+    IDNHOSTNAME = "idn-hostname"
+    IPV4 = "ipv4"
+    IPV6 = "ipv6"
+    URI = "uri"
+    URIREFERENCE = "uri-reference"
+    IRI = "iri"
+    IRIREFERENCE = "iri-reference"
+    URITEMPLATE = "uri-template"
+    JSONPOINTER = "json-pointer"
+    RELATIVEJSONPOINTER = "relative-json-pointer"
+    REGEX = "regex"
+
+
+class SecuritySchemeType(Enum):
+    APIKEY = "apiKey"
+    HTTP = "http"
+    OAUTH = "oauth2"
+    OAUTH2 = "oauth2"
+    OIDC = "openIdConnect"
+    OPENIDCONNECT = "openIdConnect"
+    MUTUALTLS = "mutualTLS"
+
+
+@dataclass
+class Contact(OpenAPIElement):
+    name: Optional[str] = None
+    url: Optional[str] = None
+    email: Optional[str] = None
+
+
+@dataclass
+class ExternalDocs(OpenAPIElement):
+    url: str
+    description: Optional[str] = None
+
+
+@dataclass
+class License(OpenAPIElement):
+    name: str
+    identifier: Optional[str] = None
+    url: Optional[str] = None
+
+
+@dataclass
+class Info(OpenAPIElement):
+    title: str
+    version: str
+    summary: Optional[str] = None
+    description: Optional[str] = None
+    terms_of_service: Optional[str] = None
+    contact: Optional[Contact] = None
+    license: Optional[License] = None
+
+
+@dataclass
+class ServerVariable(OpenAPIElement):
+    default: str
+    description: Optional[str] = None
+    enum: Optional[List[str]] = None
+
+
+@dataclass
+class Server(OpenAPIElement):
+    url: str
+    description: Optional[str] = None
+    variables: Optional[Dict[str, ServerVariable]] = None
+
+
+@dataclass
+class XML(OpenAPIElement):
+    name: Optional[str] = None
+    namespace: Optional[str] = None
+    prefix: Optional[str] = None
+    attribute: Optional[bool] = None
+    wrapped: Optional[bool] = None
+
+
+@dataclass
+class Discriminator(OpenAPIElement):
+    property_name: str
+    mapping: Optional[Dict[str, str]] = None
+
+
+@dataclass
+class Schema(OpenAPIElement):
```

Comment:
It seems we should add `content_encoding` and `content_media_type` explicitly to the Schema object.